[inductor] Enable mypy checking for torch/_inductor/bounds.py

masnesral opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #109271

Summary: Add type hints and enable mypy checking for torch/_inductor/bounds.py

Test Plan: lintrunner

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov

masnesral committed:
[inductor] Enable mypy checking for torch/_inductor/bounds.py

Summary: Add type hints and enable mypy checking for torch/_inductor/bounds.py

Test Plan: lintrunner

[ghstack-poisoned]

masnesral committed:
Update on "[inductor] Enable mypy checking for torch/_inductor/bounds.py"


Summary: Add type hints and enable mypy checking for torch/_inductor/bounds.py

Test Plan: lintrunner

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

lezcano commented on torch/_inductor/bounds.py:
```diff
@@ -54,8 +56,10 @@ def get_bounds(self):
-        result = {}
+    def swap_submodules(
+        self, submodules: Dict[str, Callable[..., Any]]
+    ) -> Dict[str, Callable[..., Any]]:
+        result: Dict[str, Callable[..., Any]] = {}
```
```suggestion
    ) -> Dict[str, Callable[..., ValueRanges]]:
        result: Dict[str, Callable[..., ValueRanges]] = {}
```

lezcano commented on torch/_inductor/bounds.py:
```diff
@@ -75,7 +79,14 @@ def swap_submodules(self, submodules):
+        mask: Any,
+        value: Any,
+        submodules: Dict[str, Callable[..., Any]],
+    ):
```
```suggestion
    ) -> ValueRanges:
```

masnesral replied:
> Some of these annotations are very informative, but others are... uf. Not a big fan of those others, but well.

@lezcano Are you commenting about the state of the types effectively already in use here or do you mean we could do better, e.g., get more detailes with the Callables? If you think we can do better, I'm up for trying harder here. I _think_ many of these were what I gathered via monkeytype, but I could certainly try again.

lezcano replied:
Oh, no, it's not your fault, it's just that many of these are rather verbose and they "do not add much information" (I'm looking at you `Callable[..., Any]`. Nothing actionable though.

I'm sorry if it sounded the wrong way :)

masnesral replied:
Oh no problem at all, just wanted to make sure I understood!

lezcano reviewed (commented):
Some of these annotations are very informative, but others are... uf. Not a big fan of those others, but well.

Just two small things.

masnesral committed:
nits on "[inductor] Enable mypy checking for torch/_inductor/bounds.py"


Summary: Add type hints and enable mypy checking for torch/_inductor/bounds.py

Test Plan: lintrunner

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 ipiszy ngimel yf225 chenyang78 kadeng muchulee8 aakhundov

[ghstack-poisoned]

lezcano commented:
@pytorchbot merge

masnesral commented:
For the test failures, I see: 

> ModuleNotFoundError: No module named 'sympy'

@lezcano , any idea what that's about?

lezcano commented:
I think that error is not there anymore, so let's try to merge this again

lezcano commented:
@pytorchbot merge
